iron-autogrow-textarea.html

```diff
@@ -324,6 +309,7 @@
         textarea.value = !(this.bindValue || this.bindValue === 0) ? '' : this.bindValue;
       }
 
+      this.value = this.bindValue;
```

Comment:
Done! I've also added a bunch of tests because I was paranoid this wasn't covering all the cases